opm/simulators/wells/WellInterfaceGeneric.hpp

```diff
@@ -301,6 +303,8 @@ class WellInterfaceGeneric {
     double well_efficiency_factor_;
     const VFPProperties* vfp_properties_;
     const GuideRate* guide_rate_;
+
+    std::vector< std::string> well_control_log_;
```

Comment:
Maybe use the control enum instead of a string here?

Reply:
> Maybe use the control enum instead of a string here?

Agreed.  Instead of spending a lot of effort formatting strings we should collect the individual transition states and only format when creating the final log statements.

Reply:
Then we will need a `well_control_log_inj_` and a `well_control_log_prod_ ` but that's fine with me.

Reply:
I will argue for keeping it std::string 1) We now need the std::string format and thus the conversion for the .debug here anyway. 2) The `reportWellSwitching` becomes more complicated if we use enum since we need to distinguish between injectors and producers.

Reply:
I think these are good arguments, ok with keeping the string.